Src/ILGPU/Runtime/ArrayViews.tt

```diff
@@ -248,7 +248,9 @@ namespace ILGPU.Runtime
 <#      for (int i = 0; i < dim.Dimension; ++i) { #>
 <#          var iName = IndexDimensions[i].PropertyName; #>
             Trace.Assert(
-                index.<#= iName #> >= 0 & index.<#= iName #> < Extent.<#= iName #>,
+                index.<#= iName #> >= 0 & (
+                    index.<#= iName #> < Extent.<#= iName #>
+                    || index.<#= iName #> == 0),
```

Comment:
nit: can we change this to a logic `|` operation in order to improve compilation speed in debug builds?